packages/config-loader/src/schema/compile.ts

```diff
@@ -102,6 +106,18 @@ export function compileConfigSchemas(
 
   const merged = mergeConfigSchemas(schemas.map(_ => _.value));
 
+  if (options?.noExtraProperties) {
```

Comment:
I feel that calling this `noAdditionalProperties` instead sticks to the established naming a bit more. Is the "extra" from somewhere specific or just shorter? 😁

Reply:
I was trying to stay away from `additionalProperties` as that's also the JSON schema wording and we're allowing `additionalProperties` when specified but enforcing it when not specified.

Reply:
Ah, you mean there might be a confusion where if we wanted to avoid this we'd need to call it something like `noAdditionalPropertiesUnlessExplicitlyAllowed`? Valid concern but doesn't feel super risky either. Felt that "extra" was basically the same thing but different word so wondered why that was the case. Do you feel `noUndeclaredProperties` would be more clear? Otherwise I'll let you pick 😁

Reply:
I like `noUndeclaredProperties`, updated.